### ISSUE TITLE
Allow add many, to defer hashing until  all nodes have been modified

### ISSUE THREAD

**Issue by inverted-capital** [rvagg/js-ipld-hashmap#30]
Thank you for this extremely useful code, and the backing implementation as well.

I have a use case where I know in advance of a batch of updates that I want, and I want to defer the serialization and hashing costs until the tree has settled.  I am happy to attempt to get this going, but this issue is here in case there are some unknown caveats I should watch out for.

**Comment by rvagg** [comment]
yeah, unfortunately "settled" requires hashing since they determine the keys, but there are costs that can be deferred and this implementation is pretty (very) inefficient at batch updates -- I'm about to start work on more efficient HAMT batching that may be of use, but unfortunately hashing is still required up-front to determine position in the tree relative to all other entries

**Comment by inverted-capital** [comment]
Thanks for writing back - yes sorry I was not specific enough - I would like to defer the hashing of parent nodes and accept that hashing is still required to generate keys.

How can we help with this work ?
